src/test/java/com/aws/greengrass/logmanager/model/LogFileTest.java

```diff
@@ -25,53 +28,86 @@ public class LogFileTest {
+        file.delete();
     }
 
+    @Test
+    void GIVEN_log_file_with_less_than_target_lines_WHEN_calculate_file_hash_THEN_we_get_null()
+            throws IOException {
+        File file = new File(directoryPath.resolve("greengrass_test.log").toUri());
+        byte[] bytesArray = getRandomTestStrings(DEFAULT_BYTES_FOR_DIGEST_NUM - 100).getBytes(StandardCharsets.UTF_8);
+        writeFiles(file, bytesArray);
+        LogFile logFile = LogFile.of(file);
+        String fileHash = logFile.hashString();
+        assertEquals(fileHash, HASH_VALUE_OF_EMPTY_STRING);
+        file.delete();
+    }
+
+
     @Test
     void GIVEN_log_file_with_equal_to_target_lines_WHEN_calculate_file_hash_THEN_we_get_null()
             throws IOException, NoSuchAlgorithmException {
         File file = new File(directoryPath.resolve("greengrass_test.log").toUri());
-        writeFiles(file, DEFAULT_LINES_FOR_DIGEST_NUM);
-        StringBuilder msg = new StringBuilder();
-        for (int i = 0; i < DEFAULT_LINES_FOR_DIGEST_NUM; i++) {
-            msg.append("line").append(i + 1);
-        }
+        byte[] bytesArray = getRandomTestStrings(DEFAULT_BYTES_FOR_DIGEST_NUM).getBytes(StandardCharsets.UTF_8);
+        writeFiles(file, bytesArray);
         LogFile logFile = LogFile.of(file);
         String fileHash = logFile.hashString();
-        assertEquals(fileHash, calculate(msg.toString()));
+        String msg = new String(bytesArray);
+        assertEquals(fileHash, calculate(msg));
         file.delete();
     }
 
     @Test
     void GIVEN_log_file_with_more_than_target_lines_WHEN_calculate_file_hash_THEN_we_get_hash()
             throws IOException, NoSuchAlgorithmException {
         File file = new File(directoryPath.resolve("greengrass_test.log").toUri());
-        writeFiles(file, DEFAULT_LINES_FOR_DIGEST_NUM + 1);
-        StringBuilder msg = new StringBuilder();
-        for (int i = 0; i < DEFAULT_LINES_FOR_DIGEST_NUM; i++) {
-            msg.append("line").append(i + 1);
-        }
+        byte[] bytesArray = getRandomTestStrings(DEFAULT_BYTES_FOR_DIGEST_NUM + 100).getBytes(StandardCharsets.UTF_8);
+        writeFiles(file, bytesArray);
+        LogFile logFile = LogFile.of(file);
+        String fileHash = logFile.hashString();
+        String msg = new String(Arrays.copyOfRange(bytesArray, 0, DEFAULT_BYTES_FOR_DIGEST_NUM));
+        assertEquals(fileHash, calculate(msg));
+        file.delete();
+    }
+
+    @Test
+    void GIVEN_log_file_with_less_than_target_lines_but_has_new_line_WHEN_calculate_file_hash_THEN_we_get_hash()
+            throws IOException, NoSuchAlgorithmException {
+        File file = new File(directoryPath.resolve("greengrass_test.log").toUri());
+        StringBuilder testString = new StringBuilder(getRandomTestStrings(DEFAULT_BYTES_FOR_DIGEST_NUM - 100));
+        testString.append(System.lineSeparator());
+        byte[] bytesArray = testString.toString().getBytes(StandardCharsets.UTF_8);
```

Comment:
I just find that reading this makes it hard to follow what you are doing we can make it simple and not be fancy like

```java
StringBuilder builder = new StringBuilder();
builder.append(getRandomTestStrings(DEFAULT_BYTES_FOR_DIGEST_NUM - 100));
builder.append(System.lineSeparator());

writeFile(file, builder.build()) // We can still pass a string here and let the write file convert it to an array
```